src/Test.EndToEnd.Baselining/Extensions/IEnumerableStringExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace SarifBaseline.Extensions
+{
+    public static class IEnumerableStringExtensions
+    {
+        /// <summary>
+        ///  Filter a set to elements which contain 'requiredPart'. If 'requiredPart' is null or empty,
+        ///  this function returns the full set as-is.
+        /// </summary>
+        /// <param name="set">IEnumerable of string to filter</param>
+        /// <param name="requiredPart">Value which must be within items to return, or null/empty for no filtering</param>
+        /// <returns>Strings in set which contain requiredPart, or full set if no requiredPart provided</returns>
+        public static IEnumerable<string> WhereContainsPart(this IEnumerable<string> set, string requiredPart)
+        {
+            if (set == null) { throw new ArgumentNullException("set"); }
+
+            if (string.IsNullOrEmpty(requiredPart))
+            {
+                return set;
+            }
+            else
+            {
+                return set.Where(path => (path?.IndexOf(requiredPart, StringComparison.OrdinalIgnoreCase) ?? -1) >= 0);
+            }
+        }
+
+        /// <summary>
+        ///  Filter an IEnumerable of string by the file extensions found on the files.
+        /// </summary>
+        /// <param name="set">Set of paths to filter</param>
+        /// <param name="fileExtension">File extension to match</param>
+        /// <returns>Items in the set which have the desired file extension</returns>
+        public static IEnumerable<string> WhereExtensionIs(this IEnumerable<string> set, string fileExtension)
+        {
+            fileExtension = "." + fileExtension.TrimStart('.');
+
+            return set.Where(path => Path.GetExtension(path).Equals(fileExtension, StringComparison.OrdinalIgnoreCase));
```

Comment:
>OrdinalIgnoreCase [](start = 100, length = 17)

This is a problem on Linux. The SDK does not systematically compare paths in a file-system-appropriate way, so I'm sure this problem occurs elsewhere. Let's file a bug on this.

Reply:
Ouch. Any setting here needs to be global in the SDK that can flow to all file-system relevant things (which are numerous). i.e., we should expect that the common case here is that any session in which the SDK is invoked is focused on processing files for one kind of file system (case-sensitive) or the other.

---
In reply to: [366544727](https://github.com/microsoft/sarif-sdk/pull/1734#discussion_r366544727) [](ancestors = 366544727)

Reply:
I filed #1736, "File path comparisons are not file-system-appropriate".